File: Tellma/ClientApp/src/app/data/dto/statement-response.ts
import { EntitiesResponse } from './entities-response';
import { DetailsEntry } from '../entities/details-entry';

export interface StatementResponse extends EntitiesResponse<DetailsEntry> {
    Opening?: number;
    Closing?: number;
    Skip?: number;
    Top?: number;
    TotalCount?: number;
}

File: Tellma/Controllers/Dto/StatementResponse.cs
using Tellma.Entities;

namespace Tellma.Controllers.Dto
{
    public class StatementResponse : EntitiesResponse<DetailsEntry>
    {
        /// <summary>
        /// Opening Balance of the result
        /// </summary>
        public decimal? Opening { get; set; }

        /// <summary>
        /// Closing Balance of the result
        /// </summary>
        public decimal? Closing { get; set; }

        public int Skip { get; set; }

        public int Top { get; set; }

        /// <summary>
        /// The total count of the result if it weren't paged
        /// </summary>
        public int TotalCount { get; set; }
    }
}
